src/render_tasks/d3d12_rtao_task.hpp

```diff
@@ -0,0 +1,292 @@
+			if (data.in_miss_shader_table[frame_idx])
+			{
+				d3d12::Destroy(data.in_miss_shader_table[frame_idx]);
+			}
+			if (data.in_hitgroup_shader_table[frame_idx])
+			{
+				d3d12::Destroy(data.in_hitgroup_shader_table[frame_idx]);
+			}
+			if (data.in_raygen_shader_table[frame_idx])
+			{
+				d3d12::Destroy(data.in_raygen_shader_table[frame_idx]);
+			}
+
+			// Set up Raygen Shader Table
+			{
+				// Create Record(s)
+				UINT shader_record_count = 1;
+				auto shader_identifier_size = d3d12::GetShaderIdentifierSize(device, data.in_state_object);
+				auto shader_identifier = d3d12::GetShaderIdentifier(device, data.in_state_object, "AORaygenEntry");
+
+				auto shader_record = d3d12::CreateShaderRecord(shader_identifier, shader_identifier_size);
+
+				// Create Table
+				data.in_raygen_shader_table[frame_idx] = d3d12::CreateShaderTable(device, shader_record_count, shader_identifier_size);
+				d3d12::AddShaderRecord(data.in_raygen_shader_table[frame_idx], shader_record);
+			}
+
+			// Set up Miss Shader Table
+			{
+				// Create Record(s)
+				UINT shader_record_count = 1;
+				auto shader_identifier_size = d3d12::GetShaderIdentifierSize(device, data.in_state_object);
+
+				auto miss_identifier = d3d12::GetShaderIdentifier(device, data.in_state_object, "MissEntry");
+				auto miss_record = d3d12::CreateShaderRecord(miss_identifier, shader_identifier_size);
+
+				// Create Table(s)
+				data.in_miss_shader_table[frame_idx] = d3d12::CreateShaderTable(device, shader_record_count, shader_identifier_size);
+				d3d12::AddShaderRecord(data.in_miss_shader_table[frame_idx], miss_record);
+			}
+
+			// Set up Hit Group Shader Table
+			{
+				// Create Record(s)
+				UINT shader_record_count = 1;
+				auto shader_identifier_size = d3d12::GetShaderIdentifierSize(device, data.in_state_object);
+
+				auto hit_identifier = d3d12::GetShaderIdentifier(device, data.in_state_object, "AOHitGroup");
+				auto hit_record = d3d12::CreateShaderRecord(hit_identifier, shader_identifier_size);
+
+				// Create Table(s)
+				data.in_hitgroup_shader_table[frame_idx] = d3d12::CreateShaderTable(device, shader_record_count, shader_identifier_size);
+				d3d12::AddShaderRecord(data.in_hitgroup_shader_table[frame_idx], hit_record);
+			}
+		}
+
+		inline void SetupAOTask(RenderSystem & render_system, FrameGraph & fg, RenderTaskHandle & handle, bool resize)
+		{
+			// Initialize variables
+			auto& n_render_system = static_cast<D3D12RenderSystem&>(render_system);
+			auto& device = n_render_system.m_device;
+			auto& data = fg.GetData<RTAOData>(handle);
+			auto n_render_target = fg.GetRenderTarget<d3d12::RenderTarget>(handle);
+			d3d12::SetName(n_render_target, L"AO Target");
+
+			if (!resize)
+			{
+				auto cmd_list = fg.GetCommandList<d3d12::CommandList>(handle);
+				auto pred_cmd_list = fg.GetPredecessorCommandList<wr::ASBuildData>();
+
+				cmd_list->m_rt_descriptor_heap = static_cast<d3d12::CommandList*>(pred_cmd_list)->m_rt_descriptor_heap;
```

Comment:
this is very old code, we changed the logic. Look at the current master version of the hybrid render task. Each command list now uses its own desc heap, we don't have to do this anymore. Also, due to this fact, the execution task in hybrid now builds the acceleration structure on the descriptor heap, therefore the AO task should do the same.

Reply:
Sould be updated